### PR TITLE
nsqadmin: make importable pkg

### PR DESCRIPTION
**Pull request by mreiferson**
pulled out of #323

RFR @jehiah

### CODE REVIEW

**Comment by mreiferson** [comment]
RFR @jehiah

File: nsqd/nsqd.go [hunk 1]
```diff
@@ -85,32 +81,13 @@ func NewNSQD(opts *nsqdOptions) *NSQD {
...
-	if err != nil {
-		n.logf("FATAL: failed to resolve HTTP address (%s) - %s", opts.HTTPAddress, err)
-		os.Exit(1)
-	}
-	n.httpAddr = httpAddr
-
-	if opts.HTTPSAddress != "" {
-		httpsAddr, err := net.ResolveTCPAddr("tcp", opts.HTTPSAddress)
+	if opts.StatsdPrefix != "" {
+		_, port, err := net.SplitHostPort(opts.HTTPAddress)
```
Comment (jehiah) [inline nsqd/nsqd.go]: looks like `opts.HTTPAddress` got rolled into the `statsdPrefix` block accidentally

File: apps/nsqadmin/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,6 @@
+FROM nsqio/nsq:latest
```
Comment (jehiah) [inline apps/nsqadmin/Dockerfile]: is this file addition intentional?

**Comment by jehiah** [comment]
LGTM asside from those two items

File: apps/nsqadmin/Dockerfile (hunk 2 above)
Comment (mreiferson) [inline apps/nsqadmin/Dockerfile]: nah, good catch (from an old rebase)

File: nsqd/nsqd.go (hunk 1 above)
Comment (mreiferson) [inline nsqd/nsqd.go]: nah, it's intentional (`statsdHostKey` needs `port`, which now comes from this line)

**Comment by mreiferson** [comment]
updated

**Pull request merged**